File: 方子樽/2022.04.26/ValidParentheses.rb
# @param {String} s
# @return {Boolean}
def is_valid(s)
    stack=[]
    s.each_char do |i|
        case i
        when '(','[','{'
            stack.push(i)
        when ')'
            return false if stack.pop()!='('
        when ']'
            return false if stack.pop()!='['
        when '}'
            return false if stack.pop()!='{'
        end
    end
    stack.empty?
end

File: 孫睦桓/2021.11.19/Valid_Parentheses.js
const checkpoint = {
    ")": "(",
    "}": "{",
    "]": "[",

};
//false:奇數個、不對稱
var isValid = function(s) {
    if (s % 2 == 1) {
        return false;
    }
    const container = [];
    for (let i = 0; i < s.length; i++) {
        const char = s[i];
        if (char == "(" || char == "[" || char == "{") {
            char.container.push();
            continue; //跳過迴圈繼續 (下一個
        }
        let last = container.pop();
        let match = checkpoint[char];
        if (last !== match) {
            return false;
        }
    }
    return stack.length === 0
};